debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/structureformat/extensions/J9ObjectStructureFormatter.java

```diff
@@ -238,6 +228,33 @@ private void formatArrayObject(PrintStream out, J9ClassPointer localClazz, U8Poi
 		}
 	}
 
+	private void formatReferenceObject(PrintStream out, int tabLevel, J9ClassPointer localClazz, int begin, int finish, J9IndexableObjectPointer array) throws CorruptDataException
+	{
+		boolean isArrayFlattened = ValueTypeHelper.getValueTypeHelper().isJ9ClassIsFlattened(localClazz);
+		int dataSize = isArrayFlattened ? J9ArrayClassPointer.cast(localClazz).flattenedElementSize().intValue() : (int)ObjectReferencePointer.SIZEOF;
+		for (int index = begin; index < finish; index++) {
+			VoidPointer slot = J9IndexableObjectHelper.getElementEA(array, index, dataSize);
+			if (slot.notNull()) {
```

Comment:
this will cause issues for valuetypes, if the first slot in a value type is null, it doesnt mean that the entire value type is null. For valueTypes the !j9object addr [#]` should always be printed.

Reply:
Does the `!fj9object 0x%x`, where %x is the compressed pointer value, have any significance for flattened arrays?

Reply:
I cant recall whether `!fj9object 0x%x` is even valid for flattened types. I think all flattened types use `j9object` so I'm not sure `fj9object addr field1.field2` will ever appear in `j9object` output.

So the short answer is no